tests/unittests/Interface/GraphContainerTest.cpp

```diff
@@ -0,0 +1,375 @@
+#include "Interface/GraphContainer.h"
+#include <gtest/gtest.h>
+
+class GraphContainerTest : public ::testing::Test {
+protected:
+  void SetUp() override {}
+  void TearDown() override {}
+};
+
+TEST_F(GraphContainerTest, adjListUndirectedUnweighted) {
+
+  // Create object of the Graph class and add edges.
+  Graph<float, 2> graph(graph::detail::GRAPH_ADJ_LIST_UNDIRECTED_UNWEIGHTED, 6);
+  graph.addEdge(0, 1);
+  graph.addEdge(1, 3);
+  graph.addEdge(1, 5);
+  graph.addEdge(1, 4);
+  graph.addEdge(2, 4);
+
+  // Print the original grpah according to the representaion.
+  std::cout << "Original Grpah: " << std::endl;
+  graph.printGraphOg();
+
+  // convert the graph to MemRef using the functions in GraphContainer.cpp and
+  // print the memref
+  auto memref2 = graph.get_Memref();
+  std::cout << "Graph in linera 2D form: " << std::endl;
+  graph.printGraph();
```

Comment:
I think there is no need to call `printGraph()` here, since it doesn't contribute directly in testing. Same suggestion for all other similar cases.